### PR TITLE
Add beatmap video support

### PR DESCRIPTION
**Pull request by EVAST9919**
Still has some issues. Pr'ing this for getting suggestions, because I'm 100% sure this is a "wrong direction"

### CODE REVIEW

File: osu.Game/Beatmaps/IWorkingBeatmap.cs [hunk 1]
```diff
@@ -25,6 +26,11 @@ public interface IWorkingBeatmap
         /// </summary>
         Texture Background { get; }
 
+        /// <summary>
+        /// Retrieves the video file for this <see cref="WorkingBeatmap"/>.
```
Comment (iiSaLMaN) [inline osu.Game/Beatmaps/IWorkingBeatmap.cs]:
```suggestion
        /// Retrieves the video background file for this <see cref="WorkingBeatmap"/>.
```

File: osu.Game/Beatmaps/WorkingBeatmap.cs [hunk 2]
```diff
@@ -183,9 +185,16 @@ public IBeatmap Beatmap
         public bool BackgroundLoaded => background.IsResultAvailable;
         public Texture Background => background.Value;
         protected virtual bool BackgroundStillValid(Texture b) => b == null || b.Available;
+
```
Comment (iiSaLMaN) [inline osu.Game/Beatmaps/WorkingBeatmap.cs]: why?

File: osu.Game/Configuration/OsuConfigManager.cs [hunk 3]
```diff
@@ -136,6 +137,7 @@ public enum OsuSetting
         DimLevel,
         BlurLevel,
         ShowStoryboard,
+        ShowVideo,
```
Comment (iiSaLMaN) [inline osu.Game/Configuration/OsuConfigManager.cs]:
```suggestion
        ShowVideoBackground,
```

**Review by frenzibyte — CHANGES_REQUESTED** [review]
This is likely not the direction we want, the proper way to support this as discussed is to move all types of backgrounds into one component that can be dimmed and blurred.

File: osu.Game/Migrations/20171019041408_InitialCreate.Designer.cs [hunk 4]
```diff
@@ -123,6 +123,8 @@ protected override void BuildTargetModel(ModelBuilder modelBuilder)
 
                     b.Property<string>("BackgroundFile");
 
+                    b.Property<string>("VideoFile");
```
Comment (peppy) [inline osu.Game/Migrations/20171019041408_InitialCreate.Designer.cs]: i don't think this needs to exist in the database

**Comment by peppy** [comment]
The database migration likely aren't needed, but however you made them is also incorrect (you somehow rewrote every migration ever). There's instructions [here](https://github.com/ppy/osu/wiki/Database-Migrations) for future reference.

File: osu.Game/Migrations/20171019041408_InitialCreate.Designer.cs (hunk 4 above)
Comment (iiSaLMaN) [reply osu.Game/Migrations/20171019041408_InitialCreate.Designer.cs]: it should? how can we know the video file of the beatmap

**Comment by peppy** [comment]
Your migration is still completely wrong. Did you not follow the instructions I gave?

**Comment by EVAST9919** [comment]
> Your migration is still completely wrong. Did you not follow the instructions I gave?

My bad, wasn't paying enough attention

**Review by peppy — APPROVED** [review]

**Pull request merged**